### PR TITLE
update complete rec

### PR DESCRIPTION
**Pull request by 876pol**
_If any of the below don't apply to this Pull Request, mark the checkbox as done._

- [ ] I have tested my code.
- [ ] I have added my solution according to the steps [here](https://usaco.guide/general/adding-solution#steps).
- [ ] I have followed the code conventions mentioned [here](https://usaco.guide/general/adding-solution/#code-conventions), which include the following: 
  - I understand that if it is clear that I have not attempted to follow these guidelines (ex. if I have not used tabs to indent), my PR will be closed.
  - If changes are requested, I will re-request the review after making the changes.

### CODE REVIEW

File: content/2_Bronze/Complete_Rec.mdx [hunk 1]
```diff
@@ -561,32 +524,29 @@ smaller than the initial string won't be generated.
 </Warning>
 
 ```cpp
-#include <iostream>
-#include <vector>
-#include <algorithm>
+#include <bits/stdc++.h>
```
Comment (SansPapyrus683) [inline content/2_Bronze/Complete_Rec.mdx:527]: why dis

File: content/2_Bronze/Complete_Rec.mdx [hunk 2]
```diff
@@ -561,32 +524,29 @@ smaller than the initial string won't be generated.
...
-	{
+	for (string perm : perms) {
 		cout << perm << endl;
 	}
 }
 ```
 
 </CPPSection>
 
+<JavaSection/>
```
Comment (SansPapyrus683) [inline content/2_Bronze/Complete_Rec.mdx:548]: what's this for

File: content/2_Bronze/Complete_Rec.mdx [hunk 3]
```diff
@@ -561,32 +524,29 @@ smaller than the initial string won't be generated.
...
-	{
+	do {
 		perms.push_back(s);
-	}
-	while (next_permutation(s.begin(), s.end()));
+	} while (next_permutation(s.begin(), s.end()));
 	cout << perms.size() << endl;
-	for (string perm : perms)
-	{
+	for (string perm : perms) {
```
Comment (SansPapyrus683) [inline content/2_Bronze/Complete_Rec.mdx:540]:
```suggestoin
	for (const string& perm : perms) {
```

File: content/2_Bronze/Complete_Rec.mdx [hunk 4]
```diff
@@ -935,23 +861,23 @@ public class Main {
 <PySection>
 
 ```py
-g = [input() for i in range(8)]
-d1 = [0] * 15
-d2 = [0] * 15
+g = [input() for _ in range(8)]
+sum_ = [False] * 15
+dif = [False] * 15
```
Comment (SansPapyrus683) [inline content/2_Bronze/Complete_Rec.mdx:866]: rename `dif` to `diff`

**Review by SansPapyrus683 — CHANGES_REQUESTED** [review]

File: content/2_Bronze/Complete_Rec.mdx (hunk 1 above)
Comment (876pol) [reply content/2_Bronze/Complete_Rec.mdx:527]: wanted to make code shorter

File: content/2_Bronze/Complete_Rec.mdx (hunk 2 above)
Comment (876pol) [reply content/2_Bronze/Complete_Rec.mdx:548]: i believe it hides the uncompleted message for java - java has no built-in method to generate permutations

**Review by SansPapyrus683 — APPROVED** [review]
i trust _all_ code has been tested?

**Comment by SansPapyrus683** [comment]
@ChuangSheep @ryanchou-dev

File: content/2_Bronze/Complete_Rec.mdx [hunk 5]
```diff
@@ -41,115 +41,101 @@ The first method would be to write a recursive function which searches over all
...
 ```
 
 </JavaSection>
-
 <PySection>
 
 ```py
 n = int(input())
-p = list(map(int, input().split()))
+weights = list(map(int, input().split()))
```
Comment (ChuangSheep) [inline content/2_Bronze/Complete_Rec.mdx:123]: why use `weights` if we used `weight` in java and cpp solutions

File: content/2_Bronze/Complete_Rec.mdx [hunk 6]
```diff
@@ -407,18 +373,15 @@ void search(string curr)
...
+	for (char c : s) {
 		char_count[c - 'a']++;
 	}
 	search("");
-	cout << perms.size() << endl;
-	for (string perm : perms)
-	{
-		cout << perm << endl;
+	cout << perms.size() << "\n";
+	for (string perm : perms) {
```
Comment (ChuangSheep) [inline content/2_Bronze/Complete_Rec.mdx:383]: optional, but I'd use reference

**Review by ChuangSheep — APPROVED** [review]
as said, if all codes are tested, then it looks fine

**Review by ryanchou-dev — APPROVED** [review]

**Comment by SansPapyrus683** [comment]
screw it, if there's problems we can always pr later

**Pull request merged**